shared/libraries/websocket_streaming/src/signal_descriptor_converter.cpp

```diff
@@ -54,30 +55,51 @@ SubscribedSignalInfo SignalDescriptorConverter::ToDataDescriptor(const daq::stre
                          "",
                          subscribedSignal.unitQuantity());
 
-        dataDescriptor.setUnit(unit);
+        dataDescriptorBuilder.setUnit(unit);
     }
 
-    dataDescriptor.setOrigin(subscribedSignal.timeBaseEpochAsString());
-    // *** meta "definition" end ***
+    dataDescriptorBuilder.setOrigin(subscribedSignal.timeBaseEpochAsString());
 
-    if (!subscribedSignal.isTimeSignal())
+    auto streamingRange = subscribedSignal.range();
+    if (streamingRange.isUnlimited())
+    {
+        // An unlimited range indicates that it is not set within the "definition" object.
+        // Since the range is used to configure the renderer, we need to set a default value for fusion
+        // (non-openDAQ) device signals.
+        // If the signal is owned by an openDAQ-enabled server, the value range, if present,
+        // is specified within the "definition" object or alternatively might be set within
+        // the "interpretation" object, if so that value will override the default one.
+        if (!subscribedSignal.isTimeSignal())
+            dataDescriptorBuilder.setValueRange(CreateDefaultRange(dataDescriptorBuilder.getSampleType()));
```

Comment:
Is "range" mandatory? I'd find it an issue of the renderer if it only accepts signals with a specified range.

Reply:
If a range is not specified for a signal, the renderer uses [default rendering limits of (0, 1)](https://github.com/openDAQ/openDAQ/blob/main/modules/ref_fb_module/src/renderer_fb_impl.cpp#L1355). Other FBs also apply their own default range if the input signal does not specify one. It might be beneficial to unify these default min/max values across various FBs.